src/commands/cmd_server.cc

```diff
@@ -826,7 +826,11 @@ class CommandScan : public CommandScanBase {
 
     ParseCursor(args[1]);
     if (args.size() >= 4) {
-      Status s = ParseMatchAndCountParam(util::ToLower(args[2]), args_[3]);
+      std::string match_flag = util::ToLower(args[2]);
```

Comment:
https://redis.io/docs/latest/commands/scan/

The third parameter from the scan command documentation can also be TYPE/COUNT. We can improve the `ParseMatchAndCountParam` instead of a temporary fix here.

My initial idea is like below:
```
struct ScanParameters {
  std::string Match;
  int64_t count;
  RedisType type; // only allowed in SCAN command
}

Status ParseScanParameters(const std::vector<std::string> &args, ScanParameters *params) {
   // ...
}
```

Reply:
@git-hulk Thank you for your CR. I have already made the necessary modifications PTAL

Reply:
@jackyyyyyssss sure, thank you!